docs/practices/ci_benchmarking.rst

```diff
@@ -0,0 +1,160 @@
+Within a couple of minutes, the changes should be reflected on the dashboard.
+
+asv-main
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This workflow is triggered on pushes to the main branch.
+
+- Computes the benchmarks for for all the commits on main for which there are no benchmark results. As such, 
+  the first run will take a while because it will process every commit on the repository to the present day, 
+  but subsequent runs will be faster.
+   
+   .. code:: bash
+
+      >> asv run ALL --skip-existing
+
+- Publishes the results to a dashboard on GitHub Pages (`<https://{{organization}}.github.io/{{project_name}}>`_).
+  
+.. note::
+   * A Github actions bot pushes the benchmark results to the main branch for future use. 
+     The workflows run consecutively to avoid any conflicts between jobs attempting to submit
+     results simultaneously. A workflow is queued up until the previous workflow running on main is finished.
+   * ASV uses the most recent benchmarking suites to compute results for the range of commits in question. 
+     Any direct change to these suites with the intent to affect the runtime or memory usage produces no 
+     change of behavior. Only changes to your project source modules do.
+
+asv-pr
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This workflow is triggered on pull requests targeting the main branch.
+
+* Compares the benchmarks of the main branch with those of main merged with the new changes.
+* Uses `asv-formatter <https://github.com/lincc-frameworks/asv-formatter>`_ to process the output.
+* Publishes a comment on the pull request with the final assessment.
+
+Below is an example of the results generated.
+
++-------------+------------+----------+------------------------------------+
+| Before      | After      | Ratio    | Method                             |
++=============+============+==========+====================================+
+| [fcd6c976]  | [bc939276] |          |                                    |
++-------------+------------+----------+------------------------------------+
+| 2.1k        | 2.1k       | 1.00     | benchmarks.MemSuite.mem_list       |
++-------------+------------+----------+------------------------------------+
+| failed      | 304±2ms    | n/a      | benchmarks.TimeSuite.time_iterkeys |
++-------------+------------+----------+------------------------------------+
+| 2.43±0.05μs | 205±0.7ms  | 84400.48 | benchmarks.TimeSuite.time_keys     |
++-------------+------------+----------+------------------------------------+
+| 9.67±0.03μs | 505±1ms    | 52177.14 | benchmarks.TimeSuite.time_range    |
++-------------+------------+----------+------------------------------------+
+| failed      | 1.01±0s    | n/a      | benchmarks.TimeSuite.time_xrange   |
++-------------+------------+----------+------------------------------------+
+
```

Comment:
Requirements question: should the pipeline fail if at least one of the benchmarks is above / below a certain threshold? Let's say 150% / 200%?

Reply:
For reference: #252